activerecord/lib/rails/generators/active_record/migration.rb

```diff
@@ -18,6 +18,8 @@ def next_migration_number(dirname)
 
       private
         def primary_key_type
+          return ", id: false" unless options[:primary_key]
+
           key_type = options[:primary_key_type]
           ", id: :#{key_type}" if key_type
```

Comment:
Couldn't you just distinguish between nil/blank and false here... have two different returns based on that.

It'd save having to provide a second option that is almost named the same and only useful for this case.

That's my preferred solution, I don't know how others feel.

Reply:
I set `true` as `options[:primary_key]` 's default value, so this'll distinguish between true and false not nil/blank and false.

Reply:
Right, but I'm saying you don't even need options[:primary_key].

You could accomplish all of this with just options[:primary_key_type], and when it's explicitly false (not just missing or nil), then we do what your code does.

Reply:
I've considered this way before, but there's another voice in my head "set `false` string  as `options[:primary_key_type]`'s value isn't a good idea". 
@rafaelfranca @kamipo  Please give some suggestions.

Reply:
I would agree with that line of thinking more if the way to define "no primary key for this table" wasn't using 'id: false'... but it is. 

As far as activerecord is concerned, false is a type of id for a table... telling it to not a type.

Reply:
Right, it should be consistent with activerecord.